Feat/fields no constraint no validation (#996)
Refinitiv/refinitiv-ui: documents/src/pages/elements/email-field.md

});
```

### Custom Validation

> For advance use cases, apps can control validation and error state of the field manually. Do not set `maxLength` & `minLength` validation constrain and set `pattern` to `''`, then validate the field and update `error` property as needed.

Reviewer: ```suggestion
For advance use cases, default validation and error state of the field can be overridden. To do this, make sure that `maxLength` & `minLength` is not set and `pattern` is set to `''`, then validate with your customised validation logic and update `error` property accordingly.
```

::

```javascript
::text-field::
const fullNameField = document.getElementById("full-name");
const emailField = document.getElementById("email");

> const responseText = document.getElementById("response-text");

Reviewer: ```suggestion
const errorText = document.getElementById("error-text");
```

Author: I insist on `response` since it shows responses not just errors.

> const save = document.getElementById("save");

Reviewer: ```suggestion
const save = document.getElementById("save-button");
```

save.addEventListener("tap", () => {

>   const incompleteError = Boolean(fullNameField.value) !== Boolean(emailField.value);

Reviewer: ```suggestion
  const isValid = Boolean(fullNameField.value) === Boolean(emailField.value);
```
```suggestion
  const isInvalid = Boolean(fullNameField.value) !== Boolean(emailField.value);
```

Author: Change to `isIncomplete` as the form could still be invalid if the provided email doesn't match the expected pattern.

Reviewer: Let's make it positive ? Something like `isComplete` maybe ?

Author: double negative could be confusing, indeed! Changed to `isPartial`

  fullNameField.error = incompleteError ? !Boolean(fullNameField.value) : false;
  emailField.error = incompleteError ? !Boolean(emailField.value) : false;
  if (incompleteError) {
    responseText.classList.add('error');

>     responseText.innerHTML = "Full name & email must be provided together";

Reviewer: ```suggestion
    responseText.textContent = "Full name & email must be provided together";
```

  } else if (emailField.value && !/^\w+@mail.com$/.test(emailField.value)) {
    emailField.error = true;
    responseText.innerHTML = "Email must be valid and end with @mail.com";
  } else {
    responseText.innerHTML = "Saved";
  }
});

const inputHandler = () => {
  responseText.classList.remove('error');
  fullNameField.error = false;
  emailField.error = false;

>   responseText.innerHTML = "<br>";

Reviewer: Can we avoid using `<br>` and use CSS to style instead

Author: done

});
```

### Custom Validation

For advance use cases, default validation and error state of the field can be overridden. To do this, make sure that `maxLength` & `minLength` are not set and `pattern` is set to `''`, then validate with your customised validation logic and update `error` property accordingly.

::

```javascript
::text-field::
const fullName = document.getElementById("full-name");
const email = document.getElementById("email");
const responseText = document.getElementById("response-text");
const save = document.getElementById("button");

save.addEventListener("tap", () => {
  const isPartial = Boolean(fullName.value) !== Boolean(email.value);

>   fullName.error = isPartial ? !Boolean(fullName.value) : false;

Reviewer 2: Redundant Boolean call

```suggestion
  fullName.error = isPartial ? !fullName.value : false;
```

>   email.error = isPartial ? !Boolean(email.value) : false;

Reviewer 2: Redundant Boolean call

```suggestion
  email.error = isPartial ? !email.value : false;
```

<ef-text-field
  id="full-name"
  aria-describedby="response-text"
  placeholder="Full name as shown on the passport">
</ef-text-field>
<label for="email">Email</label>
<ef-email-field
  id="email"
  aria-describedby="response-text"
  placeholder="Actively used email address">
</ef-email-field>
<p id="response-text"></p>
<ef-button id="button">Save</ef-button>
```

::

```html
<p>Please provide secondary contact if available</p>
<label for="full-name">Full name</label>
<ef-text-field
  id="full-name"
  aria-describedby="response-text"
  placeholder="Full name as shown on the passport">
</ef-text-field>
<label for="email">Email</label>
<ef-email-field
  id="email"
  aria-describedby="response-text"
  placeholder="Actively used email address">
</ef-email-field>
<p id="response-text"></p>
<ef-button id="button">Save</ef-button>
```

```javascript
const fullName = document.getElementById("full-name");
const email = document.getElementById("email");
const responseText = document.getElementById("response-text");
const save = document.getElementById("button");

save.addEventListener("tap", () => {
  const isPartial = Boolean(fullName.value) !== Boolean(email.value);

>   fullName.error = isPartial ? !Boolean(fullName.value) : false;

Reviewer 2: ```suggestion
  fullName.error = isPartial ? !fullName.value : false;
```

>   email.error = isPartial ? !Boolean(email.value) : false;

Reviewer 2: ```suggestion
  email.error = isPartial ? !email.value : false;
```

    responseText.classList.add('error');
    responseText.innerHTML = "Full name & email must be provided together";
  } else if (email.value && !/^\w+@mail.com$/.test(email.value)) {
    email.error = true;
    responseText.innerHTML = "Email must be valid and end with @mail.com";
  } else {
    responseText.innerHTML = "Saved";
  }
});

const inputHandler = () => {
  responseText.classList.remove('error');
  fullName.error = false;
  email.error = false;
  responseText.innerHTML = "";
};

fullName.addEventListener("input", inputHandler);
email.addEventListener("input", inputHandler);
```

```typescript
import type { TextField } from "@refinitiv-ui/elements/text-field";
import type { EmailField } from "@refinitiv-ui/elements/email-field";
import type { Button } from "@refinitiv-ui/elements/button";

const fullName = document.getElementById("full-name") as TextField;
const email = document.getElementById("email") as EmailField;
const responseText = document.getElementById("response-text") as HTMLElement;
const save = document.getElementById("button") as Button;

save.addEventListener("tap", () => {
  const isPartial = Boolean(fullName.value) !== Boolean(email.value);

>   fullName.error = isPartial ? !Boolean(fullName.value) : false;

Reviewer 2: ```suggestion
  fullName.error = isPartial ? !fullName.value : false;
```

>   email.error = isPartial ? !Boolean(email.value) : false;

Reviewer 2: ```suggestion
  email.error = isPartial ? !email.value : false;
```
